go/cmd/dolt/commands/diff.go

```diff
@@ -1534,13 +1535,73 @@ func unionSchemas(s1 sql.Schema, s2 sql.Schema) sql.Schema {
 	var union sql.Schema
 	union = append(union, s1...)
 	for i := range s2 {
-		if union.IndexOfColName(s2[i].Name) < 0 {
+		idx := union.IndexOfColName(s2[i].Name)
+		if idx < 0 {
 			union = append(union, s2[i])
+		} else {
+			// have same name. choose the one with the most 'flexible' type
+			orig := union[idx]
+			orig.Type = chooseMostFlexibleType(orig.Type, s2[i].Type)
 		}
 	}
 	return union
 }
 
+// chooseMostFlexibleType returns the type that is more 'flexible' than the other type, for the purposes of printing
+// a diff when the type of the column has changed. There are a ton of different ways we could slice this. We'll stick to
+// the following rules for the time being:
+//   - Going from any integer to a float, always take the float.
+//   - Going from a low precision float to a high precision float, we'll always take the high precision float.
+//   - Going from a low precision integer to a high precision integer, we'll always take the high precision integer.
+//     Currently, we only support this if the signage is the same.
+//   - Going from a DATE, TIME, or DATETIME to a TIMESTAMP, we'll always take the TIMESTAMP.
+//
+// If none of these rules apply, we'll just take the `a` type.
+//
+// Note this is only for printing the diff. This is not robust for other purposes.
+func chooseMostFlexibleType(a, b sql.Type) sql.Type {
+	at := a.Type()
```

Comment:
There's a `sql.Type.Promote()` function that might be useful here?
In the case of increasing a column type from int32 to int64.